backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -337,10 +339,98 @@ public Response updateEndpoint(@Context SecurityContext sec, @PathParam("id") UU
         EndpointType endpointType = endpointRepository.getEndpointTypeById(principal.getAccount(), id);
         // This prevents from updating an endpoint from system EndpointType to a whatever EndpointType
         checkSystemEndpoint(endpointType);
+
+        Endpoint ep = endpointRepository.getEndpoint(principal.getAccount(), id);
+        if (isForOpenBridge(ep)) {
+
+            // We need to save the processor name and id, as the incoming endpoint can't have them
+
+            var ref = new Object() {
+                String channel = null;
+                String processorName;
+                String processorId;
+                String url;
+
+            };
+            if (ep != null) {
+                EndpointProperties properties = ep.getProperties();
+                if (properties instanceof CamelProperties) {
+                    CamelProperties cp = (CamelProperties) properties;
+
+                    ref.processorId = cp.getExtras().get(OB_PROCESSOR_ID);
+                    ref.processorName = cp.getExtras().get(OB_PROCESSOR_NAME);
+                    CamelProperties out = endpoint.getProperties(CamelProperties.class);
+                    out.getExtras().put(OB_PROCESSOR_NAME, ref.processorName);
+                    out.getExtras().put(OB_PROCESSOR_ID, ref.processorId);
+                    ref.channel = out.getExtras().get("channel");
+                    ref.url = cp.getUrl();
+                }
+            }
+
+            try {
+                UnaryOperator<Processor> updateFun = in -> {
+                    in.getAction().getParameters().put(SLACK_CHANNEL, ref.channel);
+                    in.getAction().getParameters().put(SLACK_WEBHOOK_URL, ref.url);
+                    in.setTransformationTemplate(in.getTransformationTemplate() + " ---- updated"); // TODO remove
+                    ep.setStatus(EndpointStatus.PROVISIONING);
```

Comment:
Similarly to my two previous comments, this line could be moved out of the `UnaryOperator`.